sdk/identity/identity/src/credentials/chainedTokenCredential.ts

```diff
@@ -53,13 +56,15 @@ export class ChainedTokenCredential implements TokenCredential {
    */
   async getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken> {
     let token = null;
+    let successfulCredentialName = "";
     const errors = [];
 
     const { span, updatedOptions } = createSpan("ChainedTokenCredential-getToken", options);
 
     for (let i = 0; i < this._sources.length && token === null; i++) {
       try {
         token = await this._sources[i].getToken(scopes, updatedOptions);
+        successfulCredentialName = this._sources[i].constructor.name;
```

Comment:
Imagine I made up my own credential following the TokenCredential interface but without using a class.
Then, `this._sources[i].constructor` would be undefined

Reply:
The name will be the name of the parent class, which is probably "Object" for the case you describe.

I don't think our code is thought to work at all with credentials without a proper constructor.name. There are more places where this could be an issue.

I can log an issue, if that helps.

Reply:
I made an issue to follow up: https://github.com/Azure/azure-sdk-for-js/issues/14848